redis/src/main/scala/zio/redis/Interpreter.scala

```diff
@@ -1,135 +1,236 @@
 package zio.redis
 
-import java.io.IOException
-import java.net.{ InetSocketAddress, StandardSocketOptions }
+import java.io.{ EOFException, IOException }
+import java.net.{ InetAddress, InetSocketAddress, SocketAddress, StandardSocketOptions }
 import java.nio.ByteBuffer
-import java.nio.channels.SocketChannel
-import java.nio.charset.StandardCharsets.UTF_8
-import java.util.concurrent.atomic.AtomicBoolean
-
-import scala.collection.mutable.ArrayBuilder
+import java.nio.channels.{ AsynchronousByteChannel, AsynchronousSocketChannel, Channel, CompletionHandler }
 
 import zio._
-import zio.blocking._
+import zio.stream.Stream
 
 trait Interpreter {
+
+  import Interpreter._
+
   type RedisExecutor = Has[RedisExecutor.Service]
 
   object RedisExecutor {
 
-    val ResponseBufferSize = 1024
+    val defaultPort = 6379
 
     trait Service {
-      def execute(command: Chunk[String]): IO[RedisError, String]
+
+      def execute(command: Chunk[RespValue.BulkString]): IO[RedisError, RespValue]
+
+    }
+
+    /**
+     * This is to make better use of pipelining by writing requests to the socket in batches.
+     * Ideally this*(average RESP serialized message size) should equal the buffer size used by `ByteStream`,
+     * but we can only guesstimate.
+     */
+    private val requestQueueSize = 20
```

Comment:
Please remove the scaladoc, rename the constant to upper camel case and make it a power of two.

Reply:
What's the reason for removing Scaladoc?

Reply:
No need for scaladoc on private stuff imho. However, if you want, let's keep the remark as regular multi-line.